src/routes/safe/components/Transactions/TxList/modals/ApproveTxModal.tsx

```diff
@@ -198,212 +191,66 @@ const useTxInfo = (transaction: Props['transaction']) => {
-    txAmount: value,
-    preApprovingOwner: shouldExecute ? userAddress : undefined,
-    safeTxGas,
-    operation,
-    manualGasPrice,
-    manualMaxPrioFee,
-    manualGasLimit,
-    isExecution: willExecute,
-  })
-  const [buttonStatus] = useEstimationStatus(txEstimationExecutionStatus)
-
-  const approveTx = (txParameters: TxParameters) => {
-    if (thresholdReached && confirmations.size < _threshold) {
-      dispatch(enqueueSnackbar(NOTIFICATIONS.TX_FETCH_SIGNATURES_ERROR_MSG))
-    } else {
-      dispatch(
-        processTransaction({
-          safeAddress,
-          tx: {
-            id,
-            baseGas,
-            confirmations,
-            data,
-            gasPrice,
-            gasToken,
-            nonce,
-            operation,
-            origin,
-            refundReceiver,
-            safeTxGas,
-            safeTxHash,
-            to,
-            value,
-          },
-          userAddress,
-          notifiedTransaction: TX_NOTIFICATION_TYPES.CONFIRMATION_TX,
-          approveAndExecute: isExecution && shouldExecute && isTheTxReadyToBeExecuted,
-          ethParameters: txParameters,
-          thresholdReached,
-        }),
-      )
-    }
+  const txInfo = useTxInfo(transaction)
+  const { confirmations } = txInfo
+
+  const approveTx = (txParameters: TxParameters, delayExecution: boolean) => {
+    dispatch(
+      processTransaction({
+        safeAddress,
+        tx: txInfo,
+        userAddress,
+        notifiedTransaction: TX_NOTIFICATION_TYPES.CONFIRMATION_TX,
+        approveAndExecute: !delayExecution,
+        ethParameters: txParameters,
+        thresholdReached,
+      }),
+    )
     onClose()
   }
 
-  const getParametersStatus = () => {
-    if (isExecution || shouldExecute) {
-      return 'SAFE_DISABLED'
-    }
-
-    return 'DISABLED'
-  }
-
-  const closeEditModalCallback = (txParameters: TxParameters) => {
-    const oldGasPrice = gasPriceFormatted
-    const newGasPrice = txParameters.ethGasPrice
-    const oldGasLimit = gasLimit
-    const newGasLimit = txParameters.ethGasLimit
-    const oldMaxPrioFee = gasMaxPrioFeeFormatted
-    const newMaxPrioFee = txParameters.ethMaxPrioFee
-
-    if (oldGasPrice !== newGasPrice) {
-      setManualGasPrice(newGasPrice)
-    }
-
-    if (oldMaxPrioFee !== newMaxPrioFee) {
-      setManualMaxPrioFee(newMaxPrioFee)
-    }
-
-    if (oldGasLimit !== newGasLimit) {
-      setManualGasLimit(newGasLimit)
-    }
-  }
-
   return (
     <Modal description={description} handleClose={onClose} open={isOpen} title={title}>
-      <EditableTxParameters
-        isOffChainSignature={isOffChainSignature}
-        isExecution={willExecute}
-        parametersStatus={getParametersStatus()}
-        ethGasLimit={gasLimit}
-        ethGasPrice={gasPriceFormatted}
-        ethMaxPrioFee={gasMaxPrioFeeFormatted}
-        safeNonce={nonce.toString()}
-        safeTxGas={safeTxGas}
-        closeEditModalCallback={closeEditModalCallback}
+      <TxModalWrapper
```

Comment:
The logic of the `<TxModalWrapper />` is displaying the recommended nonce. This was one of the issues I had found previously with the refactor of this modal
![Screen Shot 2022-01-26 at 08 48 38](https://user-images.githubusercontent.com/32431609/151126216-04a42a8e-425e-4568-b208-5ca5f1cc2cc6.png)

Reply:
Will fix 👍